cmd_test.go

```diff
@@ -173,87 +195,87 @@ func TestDisabledCommands(t *testing.T) {
 
 	b.Disable([]string{"cmd"})
 	b.MessageReceived(&ChannelData{Channel: "#go-bot"}, &Message{Text: "!cmd"}, &User{Nick: "user"})
-	if len(replies) != 0 {
+
+	time.Sleep(100)
+	if len(msgs) != 0 {
 		t.Fatal("Should not execute disabled active commands")
 	}
 
 	b.Disable([]string{"passive"})
 	b.MessageReceived(&ChannelData{Channel: "#go-bot"}, &Message{Text: "regular message"}, &User{Nick: "user"})
 
-	if len(replies) != 0 {
+	time.Sleep(100)
+	if len(msgs) != 0 {
 		t.Fatal("Should not execute disabled passive commands")
 	}
 }
 
 func TestCommandNotRegistered(t *testing.T) {
-	resetResponses()
-
+	reset()
 	b := newBot()
-	b.synchronousMessageSending = true
 	b.MessageReceived(&ChannelData{Channel: "#go-bot"}, &Message{Text: "!not_a_cmd"}, &User{})
 
-	if len(replies) != 0 {
+	time.Sleep(100)
+
+	if len(msgs) != 0 {
 		t.Fatal("Should not reply if a command is not found")
 	}
 }
 
 func TestInvalidCmdArgs(t *testing.T) {
-	resetResponses()
+	reset()
 	registerValidCommand()
 
 	b := newBot()
-	b.synchronousMessageSending = true
 	b.MessageReceived(&ChannelData{Channel: "#go-bot"}, &Message{Text: "!cmd \"invalid arg"}, &User{Nick: "user"})
 
+	waitMessages(t, 1, 0)
+
 	if channel != "#go-bot" {
 		t.Error("Should reply to #go-bot channel")
 	}
-	if len(replies) != 1 {
-		t.Fatal("Invalid reply")
-	}
-	if !strings.HasPrefix(replies[0], "Error parsing") {
+	if !strings.HasPrefix(msgs[0], "Error parsing") {
 		t.Fatal("Should reply with an error message")
 	}
 }
 
 func TestErroredCmd(t *testing.T) {
-	resetResponses()
+	reset()
 	cmdError := errors.New("error")
 	RegisterCommand("cmd", "", "",
 		func(c *Cmd) (string, error) {
 			return "", cmdError
 		})
 
 	b := newBot()
-	b.synchronousMessageSending = true
 	b.MessageReceived(&ChannelData{Channel: "#go-bot"}, &Message{Text: "!cmd"}, &User{Nick: "user"})
 
+	waitMessages(t, 1, 1)
```

Comment:
I really appreciate that after this method has completed, you know that there is at least this many elements in the specified channels.